electrum/lnworker.py

```diff
@@ -1387,11 +1399,100 @@ def create_routes_for_payment(
             min_cltv_expiry,
             r_tags,
             invoice_features,
-            *, full_path: LNPaymentPath = None) -> LNPaymentRoute:
-        # TODO: return multiples routes if we know that a single one will not work
-        # initially, try with less htlcs
+            *, full_path: LNPaymentPath = None) -> Sequence[Tuple[LNPaymentRoute, int]]:
+        """Creates multiple routes for splitting a payment over the available
+        private channels.
+
+        We only use payment splitting if mpp is supported by the receiver and
+        none of the channels support the amount."""
+
+        if full_path:
+            route, amount_msat = self.create_route_for_payment(
+                amount_msat,
+                invoice_pubkey,
+                min_cltv_expiry,
+                r_tags,
+                invoice_features,
+                None,
+                full_path=full_path
+            )
+            return [(route, amount_msat)]
+
+        channels_with_funds = self.channels_with_funds()
+
+        # check if payment is supported by a single channel
+        use_single_channels = False
+        if len([v for v in channels_with_funds.values() if v > amount_msat]) > 1:
+            use_single_channels = True
```

Comment:
At this point it is too early to know whether we can pay wit a single channel. We might have a large capacity channel with a node that will be unable to forward the payment because its other channels are small. We need to run `find_route` in order to detect that. 

I think it would be simpler if single-part payments were returned in the list returned by `suggest_splits`, before multi-part payments. That way `find_route` could detect if a single part payment works, and try multi-part payments if it fails.

Also, `mpp_supported` could be passed as a parameter to `suggest_splits`, so that we don't need the if statement that  duplicates the call to `create_route_for_payment`.